src/utils.ts

```diff
@@ -360,6 +360,30 @@ export class Utils {
     }
   }
 
+  static getPositionContainerElement(el: HTMLElement): HTMLElement {
+    if (!el) return null;
+
+    const style = getComputedStyle(el);
+
+    if (style.position === 'relative' || style.position === 'absolute' || style.position === 'fixed') {
+      return el;
+    } else {
+      return Utils.getPositionContainerElement(el.parentElement);
+    }
+  }
+
+  static getContainerForPositionFixedElement(el: HTMLElement | HTMLHtmlElement): HTMLElement {

```

Comment:
don't really need HTMLHtmlElement as you are not calling root methods...

Also class could be while() loop rather then recursive (might be easier to read).